crates/heat-sdk-cli/src/cli_commands/run/training.rs

```diff
@@ -0,0 +1,199 @@
+
+    context.set_generated_crate_name("generated-heat-sdk-crate".to_string());
+
+    for backend in &args.backends {
+        for config_path in &args.configs {
+            for function in &args.functions {
+                let run_id = format!("{}", backend);
+
+                commands_to_run.push((
+                    BuildCommand {
+                        run_id: run_id.clone(),
+                        backend: backend.clone(),
+                    },
+                    RunCommand {
+                        run_id,
+                        run_params: RunParams::Training {
+                            function: function.to_owned(),
+                            config_path: config_path.to_owned(),
+                            project: args.project_path.clone(),
+                            key: args.key.clone(),
+                        },
+                    },
+                ));
+            }
+        }
+    }
+
+    let res = execute_sequentially(commands_to_run, context);
+
+    match res {
+        Ok(()) => {
+            print_info!("All experiments have run successfully!.");
+        }
+        Err(e) => {
+            return Err(anyhow::anyhow!(format!(
+                "An error has occurred while running experiments: {}",
+                e
+            )));
+        }
+    }
+
+    Ok(())
+}
+
+fn create_heat_client(api_key: &str, url: &str, wss: bool, project_path: &str) -> HeatClient {
+    let creds = HeatCredentials::new(api_key.to_owned());
+    let client_config = HeatClientConfig::builder(
+        creds,
+        ProjectPath::try_from(project_path.to_string()).expect("Project path should be valid."),
+    )
+    .with_endpoint(url)
+    .with_wss(wss)
+    .with_num_retries(10)
+    .build();
+    HeatClient::create(client_config)
+        .expect("Should connect to the Heat server and create a client")
+}
+
+fn print_available_training_functions(flags: &[Flag]) {
+    for function in flags.iter().filter(|flag| flag.proc_type == "training") {
+        print_info!("{}", format_function_flag(function));
+    }
+}
+
+fn check_function_registered(function: &str, flags: &[Flag]) -> anyhow::Result<()> {
+    let function_flags: Vec<&Flag> = flags
+        .iter()
+        .filter(|flag| flag.fn_name == function)
+        .collect();
+
+    match function_flags.len() {
+        0 => Err(anyhow::anyhow!(format!("Function `{}` is not registered as a training function. Annotate a training function with #[heat(training)] to register it.", function))),
+        1 => Ok(()),
+        _ => {
+            let function_strings: String = function_flags
+                .iter()
+                .map(|flag| format_function_flag(flag))
+                .collect::<Vec<String>>()
+                .join("\n");
+
+
+            Err(anyhow::anyhow!(format!("Function `{}` is registered multiple times. Please write the entire module path of the desired function:\n{}", function.custom_color(BURN_ORANGE).bold(), function_strings)))
```

Comment:
"Please provide the fully qualified function name by writing the entire module path of the function"